scripts/utils/trading_strategy_helpers.js

```diff
@@ -250,6 +253,60 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
     return transcript.logs[0].args.fleet
   }
 
+  /**
+   * Returns a transaction creating a (deterministically generated) fleet of Safes owned by
+   * the given master address for the given nonce.
+   *
+   * @param {Address} masterAddress address of Gnosis Safe (Multi-Sig) owning the Safes that
+   * will be created
+   * @param {number} fleetSize number of safes to be created with masterAddress as owner
+   * @param {string|BN} nonce non-repeating value that is used to determine the fleet addresses
+   * @returns {Transaction} transaction that can be used to deterministically generate a fleet
+   * with the given parameters
+   */
+  const buildDeterministicFleetOfSafes = async function (masterAddress, fleetSize, nonce) {
+    const fleetFactoryDeterministic = await fleetFactoryDeterministicPromise
+    const gnosisSafeMasterCopy = await gnosisSafeMasterCopyPromise
+
+    const data = fleetFactoryDeterministic.contract.methods
+      .deployFleetWithNonce(masterAddress, fleetSize, gnosisSafeMasterCopy.address, nonce)
+      .encodeABI()
+    const transaction = {
+      operation: CALL,
+      to: fleetFactoryDeterministic.address,
+      value: 0,
+      data,
+    }
+
+    return transaction
+  }
+
+  /**
+   * Computes a unique nonce that can be used to create a deterministic fleet deployment for
+   * the given master safe.
+   *
+   * @param {Address|SmartContract} masterSafe Gnosis Safe smart contract (or its address) that
+   * will own the fleet for which the nonce is created.
+   * @param {string} [gnosisSafeNonce=null] If set, the function uses the given nonce instead of
+   * retieving the current transaction nonce of the master safe.
+   * @returns {string} Nonce to be used on a deterministic fleet deployment.
+   */
+  const uniqueNonce = async function (masterSafe, gnosisSafeNonce = null) {
+    masterSafe = typeof masterSafe === "string" ? await getSafe(masterSafe) : masterSafe
+
+    // There can be only one transaction executed with the same Gnosis Safe nonce.
+    // This nonce can be combined with a fixed string to create a number that is
+    // unique for each deployment from the same master Safe.
+    if (gnosisSafeNonce === null) {
+      gnosisSafeNonce = await masterSafe.nonce()
+    }
+    const STRING_DESCRIPTOR = "Custom Market Maker deterministic deployment"
+    const hashInput = STRING_DESCRIPTOR + masterSafe.address + gnosisSafeNonce.toString()
```

Comment:
Yes and no, it's mostly a "bug avoiding" measure since `FleetFactoryDeterministic` could in principle also be used by other projects to deploy their safes. Without the string descriptor, the pattern would be very natural (owner + some kind of nonce) so it could inadvertently cause very annoying conflicts between unrelated projects. With the string descriptor instead there would be no such accidental hash collision.